scripts/irods/test/test_misc.py

```diff
@@ -373,3 +373,93 @@ def test_iinit_does_not_crash_from_long_home_directory__issue_5411(self):
             error_string = ' ERROR: environment_properties::capture: missing environment file. should be at [{}/.irods/irods_environment.json]\n'.format(
                 temp_home)
             self.assertEqual(stderr, error_string)
+
+
+    def test_tcp_keepalive_time__issue_2533_3824(self):
+        # This test sets the iRODS TCP keepalive time to various values. Due to the nature of TCP socket timeouts, this
+        # does not test the behavior of the TCP keepalive options on the sockets. It mainly demonstrates the range of
+        # values allowed by the system.
+        self.env = os.environ.copy()
+
+        # Test with tcp_keepalive_time unset.
+        self.admin.assert_icommand('ils', 'STDOUT', self.admin.session_collection, env=self.env)
+
+        # Test with tcp_keepalive_time set to 1. This is the smallest valid value and should not effect behavior.
+        self.env['IRODS_TCP_KEEPALIVE_TIME_IN_SECONDS'] = str(1)
+        self.admin.assert_icommand('ils', 'STDOUT', self.admin.session_collection, env=self.env)
+
+        # Test with tcp_keepalive_time set to 0. This value is invalid, but is detected by the socket-creating function
+        # and ignored, effectively leaving it unset.
+        self.env['IRODS_TCP_KEEPALIVE_TIME_IN_SECONDS'] = str(0)
+        self.admin.assert_icommand('ils', 'STDOUT', self.admin.session_collection, env=self.env)
+
+        # Test with tcp_keepalive_time set to 0. This value is invalid, but is detected by the socket-creating function
+        # and ignored, effectively leaving it unset.
+        self.env['IRODS_TCP_KEEPALIVE_TIME_IN_SECONDS'] = str(-1)
+        self.admin.assert_icommand('ils', 'STDOUT', self.admin.session_collection, env=self.env)
+
+        # Test with tcp_keepalive_time set to INT32_MAX. This value is invalid and is not detected by the
+        # socket-creating function, so it attempts to set the socket option and fails because the value is invalid.
+        # Check for SOCKET_ERROR (-179000) with an errno of 22 (Invalid argument).
+        self.env['IRODS_TCP_KEEPALIVE_TIME_IN_SECONDS'] = str(2147483647)
+        self.admin.assert_icommand('ils', 'STDERR', '-179022', env=self.env)
```

Comment:
Consider adding more of the error message to the expected output value to guard against false positives in the future. While unlikely to happen, if `-1790220` became a valid error code and was returned here, the test would accept it.

I'm assuming this actually returns something like `-179022 SOCK_ERROR`.

_Same thing for other tests._

Reply:
Done